job-scheduler/src/main/java/org/hawkular/metrics/scheduler/api/RepeatingTrigger.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2014-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.scheduler.api;
+
+import static org.hawkular.metrics.datetime.DateTimeService.getTimeSlice;
+import static org.hawkular.metrics.datetime.DateTimeService.now;
+import static org.joda.time.Duration.standardMinutes;
+
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author jsanda
+ */
+public class RepeatingTrigger implements Trigger {
+
+    private Long triggerTime;
+
+    private Long interval;
+
+    private Long delay;
+
+    private Integer repeatCount;
+
+    private Integer executionCount;
+
+    public static class Builder {
+
+        private Long interval;
+        private Long delay;
+        private Integer repeatCount;
+
+        public Builder withInterval(int interval, TimeUnit timeUnit) {
+            this.interval = TimeUnit.MILLISECONDS.convert(interval, timeUnit);
+            return this;
+        }
+
+        public Builder withDelay(int delay, TimeUnit timeUnit) {
+            this.delay = TimeUnit.MILLISECONDS.convert(delay, timeUnit);
+            return this;
+        }
+
+        public Builder withRepeatCount(int count) {
+            this.repeatCount = count;
+            return this;
+        }
+
+        public RepeatingTrigger build() {
+            return new RepeatingTrigger(interval, delay, repeatCount);
+        }
+
+    }
+
+    private RepeatingTrigger() {
+    }
+
+    private RepeatingTrigger(Long interval, Long delay, Integer repeatCount) {
```

Comment:
You would use a RepeatingTrigger. Note the logic around `repeatCount` in the `nextTrigger` method:

```java
public Trigger nextTrigger() {
    // TODO what should we do if interval is null?

    if (repeatCount != null && executionCount + 1 > repeatCount) {
        return null;
    }
    RepeatingTrigger next = new RepeatingTrigger();
    next.interval = interval;
    next.delay = delay;
    next.triggerTime = triggerTime + interval;
    next.repeatCount = repeatCount;
    next.executionCount = executionCount + 1;

    return next;
}
```